### DISCUSSION TITLE
replicating rectangle behavior / cycling floating window size

### DISCUSSION THREAD

**Discussion by clyra** [asmvik/yabai#1025] [General]
Hi there!

I'm slowing migrating from rectangle/magnet to yabai+skhd and would like to replicate one feature of rectangle app which is to cycle window size by repeating the hotkey press, i.e. if I press hyper + left, a window will occupy the half left side of the screen, you press it again it will grow to 3/4 of screen and you press again it will shrink to 1/4 of the screen and then start to cycle all again. The first resize is easy and basically you just use the --grid option, but the rest is not so easy as it would require to check if the window is already that size/ratio. Any idea how to do that? Is it possible for yabai to check if a window is already occupying a given grid cell?

**Comment by qns7** [comment]
How did you manage to get the display changing right on yabai? On my end: If I send a window to another display it goes there, but to an area far away from the actual seen screen real estate. I have to use the maximize function to move it to the seen territory. It's weird and I cannot figure it out. On the other hand I have to use exclusively the grid function to achieve the magnet/rectangle behaviour and it seems slower than these options too. I cannot give up magnet right now it seems, because it's just the most reliable, fastest and easiest way to get what I want. But I would love to just use yabai+skhd. Perhaps we can exchange ideas? Best regards Q

**Comment by Bellavene** [comment]
There is frame size in the window query output `yabai -m query --windows --window` 

```
	"frame":{
		"x":10.0000,
		"y":50.0000,
		"w":566.0000,
		"h":370.0000
	},
```

It can be scripted through jq queries, although not as easy as you wanted.
